### PR TITLE
Do not schedule targets on a revert

### PR DESCRIPTION
**Pull request by ricardoamador**
Since we do not care about waiting for targets to complete on revert requests we can ignore them and save some capacity on these requests. 

We only require the ci.yaml validation check run on reverts. Not that if we change requirements this will need to be reworked.

Another option is to mark targets as run on revert or something to that effect and let autosubmit choose which to enforce.

*List which issues are fixed by this PR. You must list at least one issue.*
Fixes https://github.com/flutter/flutter/issues/135790

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: app_dart/lib/src/service/scheduler.dart [hunk 1]
```diff
@@ -322,10 +322,14 @@ class Scheduler {
     try {
       final List<Target> presubmitTargets = await getPresubmitTargets(pullRequest);
       final List<Target> presubmitTriggerTargets = getTriggerList(presubmitTargets, builderTriggerList);
```
Comment (keyonghan) [inline app_dart/lib/src/service/scheduler.dart:324]: Nit: move these two lines to the `else` logic to skip if they come from revert.

File: app_dart/lib/src/model/appengine/commit.g.dart [hunk 2]
```diff
@@ -8,7 +8,8 @@ part of 'commit.dart';
 // JsonSerializableGenerator
 // **************************************************************************
 
-Map<String, dynamic> _$SerializableCommitToJson(SerializableCommit instance) => <String, dynamic>{
+Map<String, dynamic> _$SerializableCommitToJson(SerializableCommit instance) =>
```
Comment (keyonghan) [inline app_dart/lib/src/model/appengine/commit.g.dart:11]: Format these `.g.dart` files.

File: app_dart/lib/src/service/config.dart [hunk 3]
```diff
@@ -80,6 +80,10 @@ class Config {
     return defaultBranches[slug] ?? kDefaultBranchName;
   }
 
+  String get autosubmitBot => 'auto-submit[bot]';
+
+  static const String revertOfLabel = 'revert of';
```
Comment (keyonghan) [inline app_dart/lib/src/service/config.dart:85]: Nit: add doc for these two variables.

File: app_dart/test/service/scheduler_test.dart [hunk 4]
```diff
@@ -779,6 +779,28 @@ targets:
         );
       });
 
+      test('filters out presubmit targets that do not exist in main and do not filter targets not in main', () async {
```
Comment (keyonghan) [inline app_dart/test/service/scheduler_test.dart:782]: Update test name.

File: app_dart/lib/src/service/config.dart (hunk 3 above)
Comment (ricardoamador) [reply app_dart/lib/src/service/config.dart:85]: Gotcha.

File: app_dart/lib/src/service/scheduler.dart (hunk 1 above)
Comment (ricardoamador) [reply app_dart/lib/src/service/scheduler.dart:324]: Okay will do. I originally left this in just incase to catch formatting errors in the ci.yaml.

File: app_dart/test/service/scheduler_test.dart (hunk 4 above)
Comment (ricardoamador) [reply app_dart/test/service/scheduler_test.dart:782]: Fixed.

File: app_dart/lib/src/model/appengine/commit.g.dart (hunk 2 above)
Comment (ricardoamador) [reply app_dart/lib/src/model/appengine/commit.g.dart]: Done.

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**